### ISSUE TITLE
Mixin agent catches initialisation errors which should be fatal

### ISSUE THREAD

**Issue by Barteks2x** [SpongePowered/Mixin#344]
This makes this kind of issue much harder to diagnose than it should be, because the actual crash happens much later.

Example stacktrace:
```
[23:57:40] [main/WARN]: Mixin config cubicchunks.mixins.core.json requires mixin subsystem version 0.7.10 but 0.7.5 was found. The mixin config will not be applied.
[23:57:40] [main/ERROR]: Catching
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_221]
	at org.spongepowered.asm.launch.platform.MixinPlatformAgentFML.injectCorePlugin(MixinPlatformAgentFML.java:228) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.platform.MixinPlatformAgentFML.initFMLCoreMod(MixinPlatformAgentFML.java:155) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.platform.MixinPlatformAgentFML.<init>(MixinPlatformAgentFML.java:134) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) [?:1.8.0_221]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) [?:1.8.0_221]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) [?:1.8.0_221]
	at org.spongepowered.asm.launch.platform.MixinContainer.<init>(MixinContainer.java:68) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.platform.MixinPlatformManager.addContainer(MixinPlatformManager.java:142) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.platform.MixinPlatformManager.scanClasspath(MixinPlatformManager.java:230) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.platform.MixinPlatformManager.inject(MixinPlatformManager.java:199) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.MixinBootstrap.inject(MixinBootstrap.java:189) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.launch.MixinTweaker.injectIntoClassLoader(MixinTweaker.java:62) [liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:115) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_221]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: org.spongepowered.asm.launch.MixinInitialisationError: Required mixin config cubicchunks.mixins.core.json requires mixin subsystem version 0.7.10
	at org.spongepowered.asm.mixin.transformer.MixinConfig.checkVersion(MixinConfig.java:417) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.onLoad(MixinConfig.java:331) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:894) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:116) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:91) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:78) ~[liteloader-1.12.2-SNAPSHOT.jar:1.12.2-SNAPSHOT+jnks-b4.git-4cc2bb00c8eff08863c9f7625be08983951546a5]
	at io.github.opencubicchunks.cubicchunks.core.asm.coremod.CubicChunksCoreMod.initMixin(CubicChunksCoreMod.java:104) ~[CubicChunks-1.12.2-0.0.989.0-SNAPSHOT-all.jar:?]
	at io.github.opencubicchunks.cubicchunks.core.asm.coremod.CubicChunksCoreMod.<init>(CubicChunksCoreMod.java:75) ~[CubicChunks-1.12.2-0.0.989.0-SNAPSHOT-all.jar:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_221]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_221]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_221]
	at java.lang.Class.newInstance(Unknown Source) ~[?:1.8.0_221]
	at net.minecraftforge.fml.relauncher.CoreModManager.loadCoreMod(CoreModManager.java:594) ~[forge-1.12.2-14.23.5.2838-universal.jar:?]
	... 27 more
```

**Comment by Mumfrey** [comment]
I can't retroactively fix an issue with 0.7.5, so I can't make that issue go away. However I see where this is being caught, it's an unintentional side effect of trying to catch reflection related issues in the agent class.

**Comment by Mumfrey** [comment]
This is actually fixed in 0.8 as far as I know. Closing for now, re-open if still live.
